frontend/src/components/Forms/PartyRelationships/EditPartyRelationshipForm.js

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Field, reduxForm } from "redux-form";
+import { renderConfig } from "@/components/common/config";
+import { Form, Button, Col, Row, Popconfirm } from "antd";
+import * as FORM from "@/constants/forms";
+import { resetForm } from "@/utils/helpers";
+import EngineerOfRecordOptions from "@/components/Forms/PartyRelationships/EngineerOfRecordOptions";
+
+const propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  title: PropTypes.string.isRequired,
+  partyRelationship: PropTypes.object.isRequired,
+  mine: PropTypes.object.isRequired,
+  initialValues: PropTypes.object,
+};
+
+const defaultProps = {
+  mine: {},
+};
+
+const validate = (values) => {
+  const errors = {};
+  if (values.start_date && values.end_date) {
+    if (Date.parse(values.start_date) >= Date.parse(values.end_date)) {
+      errors.end_date = "Must be after start date.";
+    }
+  }
+  return errors;
+};
```

Comment:
Can this be kept in `utils/validate.js` to be reused in other places and for consistency

Reply:
It is a different type of validator that can't be applied to single form items and a validator for the entire form (validation logic that crosses multiple fields). It looks re-usuable in this case because I'm only checking start and end dates, but I imagine every form will need it's own set of rules and for the form validate function you cannot pass it an array of functions in the way you can for an individual field.

Reply:
ah, okay.